lib/server.js

```diff
@@ -92,6 +92,22 @@ class Server {
     } = prepareApplications(applications));
   }
 
+  // Update application version for each connection.
+  // It will update to a newest version in the requested range. If client
+  // requested version in (1.1) range he will not be updated to (1.2.x). If no
+  // suitable version is found client will use it's previous app version.
+  updateConnectionsApi() {
+    for (const connection of this.clients.values()) {
+      if (semver.valid(connection.requestedVersion)) {
+        continue;
+      }
+      const newApp = this._getApplication(
+        connection.application.name, connection.requestedVersion
+      );
+      connection.application = newApp || connection.application;
```

Comment:
In my opinion, it is better to use `if` here for readability, because in the case when `!newApp` is `true` it comes down to the assignment of the variable to itself, which isn't more clear than just doing
```javascript
if (newApp) {
  connection.application = newApp;
}
```